src/vmm/tests/integration_tests.rs

```diff
@@ -47,48 +48,83 @@ fn test_setup_serial_device() {
 
 #[test]
 fn test_build_microvm() {
-    {
-        // Error case: no boot source configured.
-        let resources: VmResources = MockVmResources::new().into();
-        let mut event_manager = EventManager::new().unwrap();
-        let empty_seccomp_filter = get_seccomp_filter(SeccompLevel::None).unwrap();
+    // Error case: no boot source configured.
+    let resources: VmResources = MockVmResources::new().into();
+    let mut event_manager = EventManager::new().unwrap();
+    let empty_seccomp_filter = get_seccomp_filter(SeccompLevel::None).unwrap();
 
-        let vmm_ret = build_microvm(&resources, &mut event_manager, &empty_seccomp_filter);
-        assert_eq!(format!("{:?}", vmm_ret.err()), "Some(MissingKernelConfig)");
-    }
+    let vmm_ret = build_microvm(&resources, &mut event_manager, &empty_seccomp_filter);
+    assert_eq!(format!("{:?}", vmm_ret.err()), "Some(MissingKernelConfig)");
+
+    // Success case.
+    // Child process will run the vmm and exit.
+    // Parent will wait for child to exit and assert on exit status 0.
+    let pid = unsafe { libc::fork() };
+    match pid {
+        0 => {
+            // Child process: build and run vmm.
+            // If the vmm thread panics, the `wait()` in the parent doesn't exit.
+            // Force the child to exit on panic to unblock the waiting parent.
+            set_panic_hook();
+
+            let boot_source_cfg: BootSourceConfig =
+                MockBootSourceConfig::new().with_default_boot_args().into();
+            let resources: VmResources = MockVmResources::new()
+                .with_boot_source(boot_source_cfg)
+                .into();
+            let mut event_manager = EventManager::new().unwrap();
+            let empty_seccomp_filter = get_seccomp_filter(SeccompLevel::None).unwrap();
 
-    {
-        // Success case.
-        let boot_source_cfg: BootSourceConfig = MockBootSourceConfig::new().with_boot_args().into();
-        let resources: VmResources = MockVmResources::new()
-            .with_boot_source(boot_source_cfg)
-            .into();
-        let mut event_manager = EventManager::new().unwrap();
-        let empty_seccomp_filter = get_seccomp_filter(SeccompLevel::None).unwrap();
+            let vmm = build_microvm(&resources, &mut event_manager, &empty_seccomp_filter).unwrap();
 
-        let vmm = build_microvm(&resources, &mut event_manager, &empty_seccomp_filter).unwrap();
-        // This exits the process, so we won't get the output from cargo.
-        vmm.lock().unwrap().stop(0);
+            // This test kernel exits after saying "Hello world". Give the vCPUs some time to exit.
+            thread::sleep(Duration::from_millis(200));
+            vmm.lock().unwrap().stop(0);
```

Comment:
If the test kernel actually exits with `HALT`/`SHUTDOWN`, then we could validate that shutdown path as well:

```suggestion
            // The vmm should exit once its workload completes and signals the exit event.
            let ev_count = event_manager.run_timeout(Duration::from_millis(500));
            // If we got here, something went wrong.
            vmm.lock().unwrap().stop(-1);
```

Reply:
The workload only completes on `x86_64`, the `aarch64` test kernel doesn't halt (there's no equivalent of `HLT` either) so I'll change this to let the kernel run its course on `x86_64` but force the vmm to stop on `aarch64`.